app/src/main/java/com/github/orkest/ui/feed/FeedActivity.kt

```diff
@@ -240,11 +249,54 @@ private fun PlayButton(song: Song){
 }
 
 @Composable
-private fun Reaction(post: Post){
+private fun LikeButton(viewModel: PostViewModel, post: Post) {
+
+    //Declaring variables to update the UI
+    val isPostLiked = remember{ mutableStateOf(false) }
+    val nbLikes = remember { mutableStateOf(post.nbLikes) }
+
+    //Fetching initial values from database
+    viewModel.isPostLiked(post).thenApply { isPostLiked.value = it }
+    viewModel.getNbLikes(post).thenApply { nbLikes.value = it }
+
+    val buttonColor = if (isPostLiked.value) Color.Yellow else Color.White
+
+    Column {
+        //Displaying the like button
+        Icon(
+            painter = painterResource(id = R.drawable.black_like_icon),
+            contentDescription = "Like button",
+            tint = buttonColor,
+            modifier = Modifier
+                .testTag("like_button")
+                .height(20.dp)
+                .width(20.dp)
+                .clickable {
+                    viewModel.updatePostLikes(post, !isPostLiked.value).thenApply {
+                        //Updating the isPostLiked value accordingly
+                        isPostLiked.value = !isPostLiked.value
```

Comment:
You should check if it completed successfully before doing so, another way to avoid that boilerplate check condition code is to make the future complete only if everything went well, otherwise it completes exceptionally and you handle the correct behavior with the whenComplete callback and the exceptional behavior with a Exceptionally callback.

You could also consider using a sealed class to handle the different states of the updatePostLikes future (success, failure, loading, etc.) and provide appropriate UI feedback to the user.